pkg/controller/migration/convert/core.go

```diff
@@ -305,38 +305,36 @@ func removeExpectedAnnotations(existing, ignoreWithValue map[string]string, toBe
 func handleNodeSelectors(c *components, install *operatorv1.Installation) error {
 	// check calico-node nodeSelectors
 	if c.node.Spec.Template.Spec.Affinity != nil {
-		if install.Spec.KubernetesProvider != operatorv1.ProviderAKS || !reflect.DeepEqual(c.node.Spec.Template.Spec.Affinity, &corev1.Affinity{
-			NodeAffinity: &corev1.NodeAffinity{
-				RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
-					NodeSelectorTerms: []corev1.NodeSelectorTerm{{
-						MatchExpressions: []corev1.NodeSelectorRequirement{{
-							Key:      "type",
-							Operator: corev1.NodeSelectorOpNotIn,
-							Values:   []string{"virtual-kubelet"},
-						}},
-					}},
-				},
-			},
-		}) {
+		var err error
+		if install.Spec.KubernetesProvider == operatorv1.ProviderAKS {
+			if aksErr := checkAKSAffinity(c.node.Spec.Template.Spec.Affinity); aksErr != nil {
```

Comment:
Similar to my validation comment, but since we support configuration of affinity with this we can just take whatever they had set previously. It worked for them before so it should still work after migration.